event/dispatcher_test.go

```
@@ -0,0 +1,106 @@
+)
+
+var ctx = context.Background()
+
+func TestDispatcher(t *testing.T) {
+	dispatcher := NewDispatcher()
+
+	t.Run("Multiple event bindings based on callback functions", func(t *testing.T) {
+		dispatcher.Reset()
+
+		dispatcher.RegisterListeners(
+			AdaptListener(ListenerFunc[*UserEvent](func(ctx context.Context, event *UserEvent) error {
```

golangci-lint (1.62.2)

[warning] unused-parameter: parameter 'ctx' seems to be unused, consider removing or renaming it as _

(revive)